pkg/controller/jobframework/reconciler.go

```diff
@@ -233,7 +235,22 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// 1. make sure there is only a single existing instance of the workload.
 	// If there's no workload exists and job is unsuspended, we'll stop it immediately.
-	wl, err := r.ensureOneWorkload(ctx, job, object)
+	var wl *kueue.Workload
+
+	if usePrebuiltWorkload {
+		wl, err = r.getPrebuiltWorkload(ctx, job, object, prebuiltWorkloadName)
+		if wl == nil {
+			if !job.IsSuspended() {
```

Comment:
Do we need this check for `!job.IsSuspended()`? I think it might happen that a job is suspended, but the nodeSelectors are not restored: https://github.com/kubernetes-sigs/kueue/blob/d91a7d5497739bf75969abb035b304d5495498b2/pkg/controller/jobs/job/job_controller.go#L160-L185. The other job types introduce the optimization of checking suspended inside the stop function.